scan_struct.go

```diff
@@ -1,17 +1,61 @@
 package pgxhelpers
 
 import (
+	"context"
 	"errors"
 	"fmt"
 	"reflect"
 	"strings"
+	"unsafe"
 
+	"github.com/jackc/pgconn"
 	"github.com/jackc/pgproto3/v2"
 	"github.com/jackc/pgx/v4"
 	"github.com/jmoiron/sqlx/reflectx"
 )
 
 var mapper = reflectx.NewMapperFunc("db", strings.ToLower)
+var rowsType reflect.Type
+
+func init() {
+	// XXX: hardcore unsafe part 1, just to make it work, not for production
+	// see part 2 below to see why I need all this
+
+	// all this unsafe black magic is required to get the instance of private *pgx.connRows for further usage
+	// to do this - need to initialise fake connection with the minimal set of fields set just to make it return
+	// instance of Rows
+
+	// pgConn is required in the connection to check connection status - we'll just set it to closed
+	pgConn := &pgconn.PgConn{}
+	pointerValPgConn := reflect.ValueOf(pgConn)
+	valPgConn := reflect.Indirect(pointerValPgConn)
+
+	fieldStatus := valPgConn.FieldByName("status")
+	ptrToStatus := unsafe.Pointer(fieldStatus.UnsafeAddr())
+	realPtrToStatus := (*byte)(ptrToStatus)
```

Comment:
It’s possible that this use of `unsafe.Pointer` is not safe. The `unsafe` package seems to be unclear as to if storing an `unsafe.Pointer` in a local variable is safe. When converting an `UnsafeAddr()` or `Pointer()` to a specific type, they seem to recommend doing it all in one expression:
```
ptrToStatus := (*byte)(unsafe.Pointer(fieldStatus.UnsafeAddr()))
```